lib/model/query/forms.js

```diff
@@ -415,6 +415,15 @@ with redacted_audits as (
     from forms
     where audits."acteeId" = forms."acteeId"
     and ${_trashedFilter(force, id, projectId, xmlFormId)}
+  ), deleted_client_audits as (
+    delete from client_audits
+    using submission_attachments, submission_defs, submissions, forms
+    where client_audits."blobId" = submission_attachments."blobId"
+    and submission_attachments."submissionDefId" = submission_defs.id
+    and submission_attachments."isClientAudit" = true
+    and submission_defs."submissionId" = submissions.id
+    and submissions."formId" = forms.id
+    and ${_trashedFilter(force, id, projectId, xmlFormId)}
```

Comment:
What if two different submission attachments share the same client audit blob data? Would this delete a client audit from a non-deleted form because it shared the same blobId? 

Is it fine if the client_audits table has something extra deleted from it because it will be processed on the fly anyway?

Reply:
This IS what happens (client audit blobs could potentially be shared and this would delete rows for a non-deleted submission). I don't like that this happens... but I don't like the other option we talked about either (having the client_audits purged after or as a result of purging a blob). My hope is that client audits are basically never going to be duplicated between submissions... they have timestamps of user actions after all, and if some client audits are deleted that shouldn't have been removed, the client audits will still export just fine via the ad-hoc processing.